kusto/ingest/internal/properties/properties.go

```diff
@@ -68,101 +72,140 @@ func (d DataFormat) CamelCase() string {
 	if err != nil {
 		return ""
 	}
+
 	return cc
 }
 
 // MarshalJSON implements json.Marshaler.MarshalJSON.
 func (d DataFormat) MarshalJSON() ([]byte, error) {
 	if d == 0 {
-		return nil, fmt.Errorf("DataFormat is an invalid compression type")
+		return nil, fmt.Errorf("DataFormat is an invalid encoding type")
 	}
+
 	return []byte(fmt.Sprintf("%q", d.String())), nil
 }
 
 const (
 	// DFUnknown indicates the EncodingType is not set.
 	DFUnknown DataFormat = 0
-	// CSV indicates the source is encoded in comma seperated values.
-	CSV DataFormat = 1
-	// JSON indicates the source is encoded in Javscript Object Notation.
-	JSON DataFormat = 2
 	// AVRO indicates the source is encoded in Apache Avro format.
-	AVRO DataFormat = 3
-	// Parquet indicates the source is encoded in Apache Parquet format.
-	Parquet DataFormat = 4
+	AVRO DataFormat = 1
+	// APACHEAVRO indicates the source is encoded in Apache avro2json format.
+	APACHEAVRO DataFormat = 2
+	// CSV indicates the source is encoded in comma seperated values.
+	CSV DataFormat = 3
+	// JSON indicates the source is encoded as one or more lines, each containing a record in Javscript Object Notation.
+	JSON DataFormat = 4
+	// MULTIJSON indicates the source is encoded in JSON-Array of individual records in Javscript Object Notation.
+	MULTIJSON DataFormat = 5
 	// ORC indicates the source is encoded in Apache Optimized Row Columnar format.
-	ORC DataFormat = 5
+	ORC DataFormat = 6
+	// Parquet indicates the source is encoded in Apache Parquet format.
+	Parquet DataFormat = 7
 	// PSV is pipe "|" separated values.
-	PSV DataFormat = 6
+	PSV DataFormat = 8
 	// Raw is a text file that has only a single string value.
-	Raw DataFormat = 7
+	Raw DataFormat = 9
 	// SCSV is a file containing semicolon ";" separated values.
-	SCSV DataFormat = 8
+	SCSV DataFormat = 10
 	// SOHSV is a file containing SOH-separated values(ASCII codepont 1).
-	SOHSV DataFormat = 9
-	// TSV is a file containing table seperated values ("\t").
-	TSV DataFormat = 10
+	SOHSV DataFormat = 11
+	// SINGLEJSON indicates the source containg a single Javascript Object Notation encoded record, newlines are treated as whitespace
+	SINGLEJSON DataFormat = 12
+	// SSTREAM indicats the source is encoded as a Microsoft Cosmos Structured Streams format
+	SSTREAM DataFormat = 13
+	// TSV is a file containing tab seperated values ("\t").
+	TSV DataFormat = 14
+	// TSVE is a file containing escaped-tab seperated values ("\t").
+	TSVE DataFormat = 15
 	// TXT is a text file with lines deliminated by "\n".
-	TXT DataFormat = 11
+	TXT DataFormat = 16
+	// WCLOGFILE indicates the source is encoded using W3C Extended Log File format
+	WCLOGFILE DataFormat = 17
 )
 
+type dataFormatDescriptior struct {
```

Comment:
names of all attributes should start lower case: aka camelCaseName, ....

Also, shorten names:

camelName
jsonName
ext
mappingReq